File: Leetcode/0435 Non-overlapping Intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, Comparator.comparingInt(interval -> interval[0]));
        
        int lastEnd = Integer.MIN_VALUE, removedInterval = 0;
        for (var interval : intervals) {
            boolean isOverlapping = interval[0] < lastEnd;
            if (isOverlapping) {
                lastEnd = Math.min(interval[1], lastEnd);
                ++removedInterval;
            } else {
                lastEnd = interval[1];
            }
        }
        return removedInterval;
    }
}

File: Leetcode/0435 Non-overlapping Intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals):
        intervals.sort()

        last_end, removed_interval = float("-inf"), 0
        for start, end in intervals:
            is_overlapping = start < last_end
            if is_overlapping:
                last_end = min(end, last_end)
                removed_interval += 1
            else:
                last_end = end

        return removed_interval